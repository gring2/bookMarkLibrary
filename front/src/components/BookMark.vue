<template>
  <div>
    <img :src="bookmark.img" alt="">
    <a :href="bookmark.url">{{bookmark.name}}</a>
  </div>
</template>

<script lang="ts">
  import {Component, Vue, Prop} from 'vue-property-decorator'
  import BookMarkModel from '@/vo/BookMark'

  @Component({

  })
  export default class BookMark extends Vue {
    @Prop({default: () =>  new BookMarkModel('', '' ,'')}) private bookmark!: BookMarkModel

    private search() {
      console.log('search')
    }
  }
</script>

<style scoped lang="scss">
  $defaultImgSize : 7rem;
  div{
    width: $defaultImgSize;
    height: $defaultImgSize;
    img{
      width: inherit;
      height: $defaultImgSize - 1rem;
    }
    a{
      font-size: 1rem;
    }
  }

</style>
