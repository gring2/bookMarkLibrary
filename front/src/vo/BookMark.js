export default class BookMark {
    constructor(title, url, thumbnail) {
        this.title = title;
        this.url = url;
        this.thumbnail = thumbnail;
    }
}
//# sourceMappingURL=BookMark.js.map