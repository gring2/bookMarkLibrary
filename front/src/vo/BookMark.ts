export default class BookMark {
  public title: string
  public url: string
  public thumbnail: string
  constructor(title: string, url: string, thumbnail: string) {
    this.title = title
    this.url = url
    this.thumbnail = thumbnail
  }

}