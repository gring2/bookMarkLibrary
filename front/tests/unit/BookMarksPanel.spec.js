import { mount } from '@vue/test-utils';
import BookMarksPanel from '@/views/BookmarksPanel.vue';
import BookMark from '@/vo/BookMark';
describe('BookMarksPenel Test', () => {
    it('render BookMarksPenel view', () => {
        const wrapper = mount(BookMarksPanel, {
            data: () => {
                return {
                    bookmarks: [new BookMark('tes_title1', 'test_url1', 'test_thumbnail1'),
                        new BookMark('tes_title2', 'test_url2', 'test_thumbnail2')
                    ]
                };
            }
        });
        expect(wrapper.text().indexOf('tes_title1')).toBeTruthy();
        expect(wrapper.text().indexOf('test_url1')).toBeTruthy();
        expect(wrapper.text().indexOf('test_thumbnail1')).toBeTruthy();
        expect(wrapper.text().indexOf('test_url2')).toBeTruthy();
        expect(wrapper.text().indexOf('tes_title2')).toBeTruthy();
        expect(wrapper.text().indexOf('test_thumbnail2')).toBeTruthy();
        expect(wrapper).toMatchSnapshot();
    });
});
//# sourceMappingURL=BookMarksPanel.spec.js.map