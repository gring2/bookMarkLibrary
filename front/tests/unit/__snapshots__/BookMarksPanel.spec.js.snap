// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookMarksPenel Test render BookMarksPenel view 1`] = `
<main>
  <div><input type="text" placeholder="url" style="width: 100%; font-size: 2rem; height: 2rem;"> <input type="text" placeholder="#tag" style="width: 100%; font-size: 2rem; height: 2rem;"> <button class="submit" style="font-weight: normal; font-size: 1.4rem; width: 8rem;">Susbmit</button></div>
  <div class="container">
    <ul>
      <li>
        <div><img src="https://cdn.qiita.com/assets/qiita-fb-2887e7b4aad86fd8c25cea84846f2236.png" alt=""> <a href="#">title</a></div>
      </li>
    </ul>
  </div>
</main>
`;
